gentoopm/portagepm/pkg.py

```diff
@@ -322,6 +329,38 @@ def maintainers(self):
 
         return tuple(PortagePackageMaintainer(m) for m in meta.maintainers())
 
+    @property
+    def upstream(self) -> PMUpstream | None:
+        xml_path = Path(self.path).parent / "metadata.xml"
+        try:
+            meta = MetaDataXML(xml_path, None)
+        except (IOError, OSError) as e:
+            if e.errno == errno.ENOENT:
+                return None
+            raise e
+
+        upstreams = meta.upstream()
+        if len(upstreams) == 0:
+            return None
+        upstream = upstreams[0]
+
+        result = PMUpstream()
+        if len(upstream.bugtrackers) != 0:
+            result.bugs_to = upstream.bugtrackers[0]
+        if len(upstream.changelogs) != 0:
+            result.changelog = upstream.changelogs[0]
+        if len(upstream.docs) != 0:
+            result.doc = upstream.docs[0]
```

Comment:
Does this handle l10n quasi-correctly (i.e. return the English version)?

Reply:
New `PMUpstreamDoc` type handles this